src/main/java/org/opentripplanner/middleware/controllers/api/ApiController.java

```diff
@@ -30,57 +29,106 @@
+    public ApiController (String apiPrefix, TypedPersistence<T> persistence) {
         this.clazz = persistence.clazz;
         this.persistence = persistence;
         this.classToLowercase = persistence.clazz.getSimpleName().toLowerCase();
         this.ROOT_ROUTE = apiPrefix + SECURE + classToLowercase;
-        registerRoutes();
     }
 
     /**
-     * Register basic CRUD HTTP endpoints for the controller implementation.
+     * This method is called by spark-swagger to register endpoints and generate the docs.
+     * @param restApi The object to which to attach the documentation.
      */
-    private void registerRoutes() {
-        LOG.info("Registering routes for {}", ROOT_ROUTE);
-        // Options response for CORS
-        options(ROOT_ROUTE, (q, s) -> "");
-        // Get multiple entities.
-        get(ROOT_ROUTE, this::getMany, JsonUtils::toJson);
-        // Get one entity.
-        get(ROOT_ROUTE + ID_PARAM, this::getOne, JsonUtils::toJson);
-        // Create entity request
-        post(ROOT_ROUTE, this::createOrUpdate, JsonUtils::toJson);
-        // Update entity request
-        put(ROOT_ROUTE + ID_PARAM, this::createOrUpdate, JsonUtils::toJson);
-        // Delete entity request
-        delete(ROOT_ROUTE + ID_PARAM, this::deleteOne, JsonUtils::toJson);
+    @Override
+    public void bind(final SparkSwagger restApi) {
+        LOG.info("Registering routes and enabling docs for {}", ROOT_ROUTE);
+
+        restApi.endpoint(endpointPath(ROOT_ROUTE)
+                .withDescription("Interface for querying and managing '" + classToLowercase + "' entities."), (q, a) -> LOG.info("Received request for '{}' Rest API", classToLowercase))
+
+
+                // If using lambdas with the GET method, a bug in the documentation framework
+                // requires you to write path(<entire_route>).
+                // If you use new GsonRoute() {...} with the GET method, you only need to write path(<relative_to_endpointPath>).
+                // Other HTTP methods are not affected by this bug.
+
+                // Get multiple entities.
+                // get(ROOT_ROUTE, this::getMany, JsonUtils::toJson);
+                .get(path(ROOT_ROUTE)
+                        .withDescription("Gets a list of all '" + classToLowercase + "' entities.")
+                        .withResponseAsCollection(clazz),
+                        this::getMany, JsonUtils::toJson
+                )
+
+                // Get one entity.
+                // get(ROOT_ROUTE + ID_PARAM, this::getOne, JsonUtils::toJson);
+                .get(path(ROOT_ROUTE + ID_PARAM)
+                        .withDescription("Returns a '" + classToLowercase + "' entity with the specified id, or 404 if not found.")
+                        .withPathParam().withName("id").withDescription("The id of the entity to search.").and()
+                        // .withResponses(...) // FIXME: not implemented (requires source change).
```

Comment:
What does withResponses do? Would we need it here? If it would be useful (now or in the future), please expand your comment a little about why.

Reply:
`withResponses` is supposed to document the expected HTTP responses (200, 403, 404)... but that doesn't appear to be implemented in spark-swagger.